evm/src/cpu/kernel/context_metadata.rs

```diff
@@ -0,0 +1,37 @@
+/// These metadata fields contain VM state specific to a particular context.
+#[derive(Copy, Clone, Eq, PartialEq, Hash, Ord, PartialOrd, Debug)]
+pub(crate) enum ContextMetadata {
+    /// The ID of the context which created this one.
+    ParentContext = 0,
+    /// The program counter to return to when we return to the parent context.
+    ParentProgramCounter = 1,
+    /// If we're in a system call, this holds the userspace program counter to return to.
+    UserspaceProgramCounter = 2,
```

Comment:
Oh, in #653 I put put that on the stack

Reply:
This also permits nested syscalls/traps.

Reply:
Ah right, forgot we decided that. Removed this.